### ISSUE TITLE
TCP input plugin failing to receive large lines

### ISSUE THREAD

**Issue by joseliber** [fluent/fluent-bit#1702]
## Bug Report

**Describe the bug**
My log files have some lines which are very large (e.g. one of the lines in my testing sample has ~10 Megabytes). When the TCP plugin receives the log lines, it fails to parse the large lines correctly, giving me just the last part of the line (probably because it is the part that contains the "\n").

**To Reproduce**
- Start fluent-bit with the TCP plugin and a chunk_size setting that is larger than the message you're going to send.
- Send the large message (between 5 - 10 Megabyte) to fluent-bit, check with strace, stdout plugin.
- Example log message if applicable:
```
[date] some_small_text_here  {"name":"a_name","hash":"small","body":"large_text..."}
[date] some_small_text_here      {"batch":"string_here","invoices": large_lots_of_important_info...}
```

**Expected behavior**
I expect fluent-bit to receive the full message, according to the size set in chunk_size.

**Screenshots**
Strace output showing the chunk_size correctly applied:
![image](https://user-images.githubusercontent.com/55399037/68025293-2f8e0100-fc8b-11e9-8d64-fdf948d876f7.png)

Fluent-bit output plugin (it should have a [date] at the beginning of the log):
![image](https://user-images.githubusercontent.com/55399037/68025578-f4d89880-fc8b-11e9-8e41-f3db73c3ff16.png)

**Your Environment**

* Version used: Fluent-bit docker image tagged 1.3.2, ID d0e68c52f3ae
* Configuration of the input plugin:
```
[INPUT]
    Name                    tcp
    Listen                  0.0.0.0
    Port                    5170
    Format                  none
    Chunk_Size              32000
    Separator               \n
```
* Environment name and version: docker 1:19.03.4
* Operating System and version: Arch Linux 5.3.7-arch1-2-ARCH
* Filters and plugins: TCP input, stdout output

**Additional context**

I'm trying to use fluent-bit to receive messages via TCP, parse them to GELF format and send them to my Graylog server.
I'm testing the functionality with a 20 Megabyte log file, but in production systems the logs will have > 400 Megabytes, so I expect to find more large lines.

Here is a sample file with ~8 Megabytes:
[largefile.zip](https://github.com/fluent/fluent-bit/files/3798649/largefile.zip)

**Comment by fujimotos** [comment]
I can confirm this is broken in HEAD: Sending a very long line to `in_tcp`
discards the reading part of the data, which results in a truncated output.

For instance, the following command:

    printf "%*s\n" 1048576  > /dev/tcp/127.0.0.1/5170

... produces the following trace log:

    [2019/11/05 14:12:03] [trace] [in_tcp] new TCP connection arrived FD=19
    [2019/11/05 14:12:03] [trace] [in_tcp] read()=8192 pre_len=0 now_len=8192
    [2019/11/05 14:12:03] [trace] [in_tcp] read()=8192 pre_len=0 now_len=8192
    [2019/11/05 14:12:03] [trace] [in_tcp] read()=4096 pre_len=0 now_len=4096
    ...
    [2019/11/05 14:12:03] [trace] [in_tcp] fd=19 closed connection

Evidently Fluent Bit incorrectly overwrites `conn->buf` on each read (see the
previous read length `pre_len` is always set to zero). My suspicion is that
`in_tcp`'s read callback is failing to update the position marker correctly.

**Comment by fujimotos** [comment]
@joseliber Should be fixed by #1713. Try the latest HEAD
of fluent-bit and see if it works as you expect.
